src/new_index/schema.rs

```diff
@@ -270,6 +269,149 @@ impl Indexer {
+        self.cleanup_cache(&orphaned_heights)?;
+
+        Ok(())
+    }
+
+    /// Clean up history entries for specific orphaned heights
+    fn cleanup_history(&self, orphaned_heights: &HashSet<usize>) -> Result<()> {
+        let _timer = self.start_timer("reorg_cleanup_history");
+        let mut batch = rocksdb::WriteBatch::default();
+        let mut count = 0;
+
+        // Scan history entries (scripthash history with 'H' prefix)
+        for row in self.store.history_db.iter_scan(&[b'H']) {
+            let history_row = TxHistoryRow::from_row(row);
+            let height = history_row.key.confirmed_height as usize;
+
+            if orphaned_heights.contains(&height) {
+                batch.delete(&history_row.into_row().key);
+                count += 1;
+            }
+        }
+
+        // Also clean up asset history entries if on Elements/Liquid
+        #[cfg(feature = "liquid")]
+        for row in self.store.history_db.iter_scan(&[b'I']) {
+            let history_row = TxHistoryRow::from_row(row);
+            let height = history_row.key.confirmed_height as usize;
+
+            if orphaned_heights.contains(&height) {
+                batch.delete(&history_row.into_row().key);
+                count += 1;
+            }
+        }
+
+        info!("Deleted {} orphaned history entries", count);
+        self.store.history_db.write_batch(batch);
+        Ok(())
+    }
+
+    /// Clean up confirmation entries for specific orphaned blockhashes
+    fn cleanup_confirmations(&self, orphaned_hashes: &HashSet<BlockHash>) -> Result<()> {
+        let _timer = self.start_timer("reorg_cleanup_confirmations");
+        let mut batch = rocksdb::WriteBatch::default();
+        let mut count = 0;
+
+        // Scan confirmation entries (prefix 'C')
+        for row in self.store.txstore_db.iter_scan(&[b'C']) {
+            let conf_row = TxConfRow::from_row(row);
+            let blockhash: BlockHash = deserialize(&conf_row.key.blockhash).unwrap();
+
+            if orphaned_hashes.contains(&blockhash) {
+                batch.delete(&conf_row.into_row().key);
+                count += 1;
+            }
+        }
+
+        info!("Deleted {} orphaned confirmation entries", count);
+        self.store.txstore_db.write_batch(batch);
+        Ok(())
+    }
+
+    /// Clean up cached data for specific orphaned heights
+    fn cleanup_cache(&self, orphaned_heights: &HashSet<usize>) -> Result<()> {
+        let _timer = self.start_timer("reorg_cleanup_cache");
+        let mut batch = rocksdb::WriteBatch::default();
+        let mut count = 0;
+
+        // Clean up aggregated stats (prefix 'A')
+        for row in self.store.cache_db.iter_scan(&[b'A']) {
+            let key = row.key;
+            // AggStats keys contain height
+            // The key format is: b'A' + scripthash + height (big-endian u32)
+            if key.len() >= 37 {
+                let height_bytes = &key[33..37];
```

Comment:
This makes the code dependent on the exact layout of keys, which would break if we ever changed them. I would instead deserialize into the db `*Key` structs and get the height from there.